### ISSUE TITLE
Address Han Unification for CJK languages in Editor

### ISSUE THREAD

**Issue by SaracenOne** [godotengine/godot-proposals#3766]
### Describe the project you are working on

General editor improvements

### Describe the problem or limitation you are having in your project

It has come to my attention that there may be issues with the way Godot renders text between the Japanese, Korean, Traditional Chinese and Simplified Chinese localization. This problem is known as Han Unification, and is a result of these different languages rendering glyphs which the Unicode consortium has assigned matching codepoints, but which these languages will sometimes render in vastly different ways. Description of these problems can be seen here.
       
https://en.wikipedia.org/wiki/Han_unification
https://heistak.github.io/your-code-displays-japanese-wrong/

Since I am not a native speaker of these languages, I would like to open this proposal as forum for anyone who is more familiar with these languages to discuss the issue, how it might be addressed differently, and just how much of a priority it should even be considered. My understanding is that this issue does concern overall ability for people to understand Godot in these languages, but does make it appear somewhat sloppy from a localisation perspective. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Since currently all CJK languages use the same font for glyph rendering, my proposal is to take advantage of improvements to Godot’s text rendering to selectively render different fonts for with associated languages, similar to how the web does, but provide the editor with the fonts necessary to render each of these languages correctly.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I have submitted an initial draft PR (https://github.com/godotengine/godot/pull/56558) which may address this issue by modifying the build system to include all the CJK fonts from the Noto set and assign them the appropriate languages which should allow them to be selectively rendered correctly depending on the chosen language code. This appears to be the most straightforward solution to the problem, but it unfortunately seems like it would result in potentially undesired bloat to the Godot binary size. A solution involving downloadable 'localisation packs' might be a more scalable alternative solution, but would require more engineering to properly implement.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

While it is perfectly possible for users to install their own editor fonts to work around this issue, this proposal is an attempt to address out of the box experience of starting the editor with these languages chosen. A user may not be aware of the intracies of this problem to install an alternative editor font.

### Is there a reason why this should be core and not an add-on in the asset library?

I have proposed the possibility of 'localisation packs' as a more scalable solution to addressing the problem than simply embedding more fonts into the binary, but currently, the editor lacks the infrastructure required to support localisation packs as a feature.

**Comment by timothyqiu** [comment]
Being able to load a system font might also be an option: #306 

I noticed this issue since my first day with Godot. The 路径 (path) on Project Manager is definitely rendered with a Japanese font. Most of these glyph differences won't bring trouble understanding the word I think (like color and colour). But they do make the interface feel like it's in the 1980s :rofl:

**Comment by SaracenOne** [comment]
Yeah, I think it's something non-native users of these languages can't really understand, but the blog post I was linked really helped contextualize how sloppy this must look. It seems a sensible time now to address this since Godot 4.0 has made the nessecary underlying changes to its text engine to finally 'fix' this in the same way the web 'fixes' it. It seems Han Unification was always a controversial issue.

Incidentally, if I manage to get one more CJK-specific localisation improvement in, it'll be further improving IME integration with inlined composition. I've almost got this working on Windows, but the APIs for integrating IMEs are kind of horrific. I'm hoping that making the localisations as high quality as possible may introduce new audience to Godot who haven't tried it yet.

**Comment by Calinou** [comment]
> Being able to load a system font might also be an option: #306

I think this is the ideal option as it will also decrease the editor's binary size. It also allows for loading emoji fonts as a fallback to ensure that emoji can be displayed in the editor :slightly_smiling_face:

**Comment by Calinou** [comment]
Now that the editor can load system fonts, this should be able to be implemented without increasing the editor's binary size (on desktop platforms only).

**Comment by timothyqiu** [comment]
The editor now tries to load system Noto Sans CJK fonts. This is how the text `骨返刃` is displayed in the editor on my machine:

| `zh_CN` | `ja` | `ko` |
|----|----|----|
|![Image](https://github.com/user-attachments/assets/e25df7f2-4dc7-4dd0-884b-f6daca15ade2) | ![Image](https://github.com/user-attachments/assets/d2dfd579-14ce-4828-8938-8dc35a12efde) |![Image](https://github.com/user-attachments/assets/6a7582ad-267d-4685-aa05-992b8186654e) |
